static/js/components/HomePage.jsx.template

```diff
@@ -104,60 +102,63 @@ const HomePage = () => {
   };
 
   return (
-    <div>
-      <ResponsiveGridLayout
-        className="layout"
-        layouts={currentLayouts}
-        breakpoints={{ '{' + app.homepage_grid.breakpoints|tojson + '}' }}
-        cols={{ '{' + app.homepage_grid.cols|tojson + '}' }}
-        margin={[10, 10]}
-        onLayoutChange={LayoutChangeHandler}
-        draggableHandle=".dragHandle"
-        rowHeight={gridRowHeight}
-      >
-        {%- for widgetName, widgetProps in app.homepage_widgets.items() %}
-          {%- if not (widgetProps.show is defined and not widgetProps.show) %}
-            {%- if widgetName == "GroupList" %}
-        <div key="{{ widgetName }}">
-          <{{ widgetName }}
-            groups={groups}
-            classes={classes}
-            {%- for key, value in widgetProps.props.items() -%}
-            {%- if value is string %}
-            {{ key }}={"{{ value }}"}
-            {%- else -%}
-            {{ key }}={ {{ value }} }
-            {%- endif -%}
-            {%- endfor %}
-          />
-        </div>
-            {%- else %}
-        <div key="{{ widgetName }}">
-          <{{ widgetName }}
-            classes={classes}
-            {%- if widgetProps.props is defined %}
-            {% for key, value in widgetProps.props.items() -%}
-            {%- if value is string %}
-            {{ key }}={"{{ value }}"}
-            {%- else -%}
-            {{ key }}={ {{ value }} }
+    preferences &&
+    Object.keys(preferences).length !== 0 && (
```

Comment:
I think that's probably also why all these tests are failing too.

Reply:
Okay @acrellin, I've now changed it to use `profile.username` instead to check that the profile has been fetched from the database since all users should have the `username` populated. This way, empty preferences will be populated with the default layout as desired, without the defaults also overwriting existing preferences while the `GET /api/internal/profile` call is in progress.